ccl/multiregionccl: TestAlterTableLocalityRegionalByRowCorrectZoneConfigBeforeBackfill failed

ccl/multiregionccl.TestAlterTableLocalityRegionalByRowCorrectZoneConfigBeforeBackfill [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsGcpLinuxX8664BigVm_CclUnitTests/12499373?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsGcpLinuxX8664BigVm_CclUnitTests/12499373?buildTab=artifacts#/) on release-23.2 @ [4b9b072a1dc54be28deffd366bc145a7d749515d](https://github.com/cockroachdb/cockroach/commits/4b9b072a1dc54be28deffd366bc145a7d749515d):


```
=== RUN   TestAlterTableLocalityRegionalByRowCorrectZoneConfigBeforeBackfill
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/96442b76c285f60de3ae793a1f7d862e/logTestAlterTableLocalityRegionalByRowCorrectZoneConfigBeforeBackfill937616132
    test_log_scope.go:81: use -show-logs to present logs inline
    regional_by_row_test.go:167: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/96442b76c285f60de3ae793a1f7d862e/logTestAlterTableLocalityRegionalByRowCorrectZoneConfigBeforeBackfill937616132
--- FAIL: TestAlterTableLocalityRegionalByRowCorrectZoneConfigBeforeBackfill (37.60s)
=== RUN   TestAlterTableLocalityRegionalByRowCorrectZoneConfigBeforeBackfill/REGIONAL_BY_ROW_to_GLOBAL
    test_server_shim.go:152: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
    alter_primary_key.go:105: pq: setting sql.virtual_cluster.feature_access.multiregion.enabled disallows use of multi-region abstractions
    --- FAIL: TestAlterTableLocalityRegionalByRowCorrectZoneConfigBeforeBackfill/REGIONAL_BY_ROW_to_GLOBAL (7.54s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #113654 ccl/multiregionccl: TestAlterTableLocalityRegionalByRowCorrectZoneConfigBeforeBackfill failed [C-test-failure O-remote-execution O-robot T-sql-foundations branch-master release-blocker]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAlterTableLocalityRegionalByRowCorrectZoneConfigBeforeBackfill.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-33179

ccl/multiregionccl.TestAlterTableLocalityRegionalByRowCorrectZoneConfigBeforeBackfill [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12544236?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12544236?buildTab=artifacts#/) on release-23.2 @ [56fd045302ed2edc14831fb001ef77486ae8f62a](https://github.com/cockroachdb/cockroach/commits/56fd045302ed2edc14831fb001ef77486ae8f62a):


```
=== RUN   TestAlterTableLocalityRegionalByRowCorrectZoneConfigBeforeBackfill
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/e558fc8050776f4c54ea39ba371b49da/logTestAlterTableLocalityRegionalByRowCorrectZoneConfigBeforeBackfill3590818859
    test_log_scope.go:81: use -show-logs to present logs inline
    regional_by_row_test.go:167: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/e558fc8050776f4c54ea39ba371b49da/logTestAlterTableLocalityRegionalByRowCorrectZoneConfigBeforeBackfill3590818859
--- FAIL: TestAlterTableLocalityRegionalByRowCorrectZoneConfigBeforeBackfill (33.31s)
=== RUN   TestAlterTableLocalityRegionalByRowCorrectZoneConfigBeforeBackfill/GLOBAL_to_REGIONAL_BY_ROW
    test_server_shim.go:152: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
    alter_primary_key.go:105: pq: setting sql.virtual_cluster.feature_access.multiregion.enabled disallows use of multi-region abstractions
    --- FAIL: TestAlterTableLocalityRegionalByRowCorrectZoneConfigBeforeBackfill/GLOBAL_to_REGIONAL_BY_ROW (6.01s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #113654 ccl/multiregionccl: TestAlterTableLocalityRegionalByRowCorrectZoneConfigBeforeBackfill failed [C-test-failure O-remote-execution O-robot T-sql-foundations branch-master release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAlterTableLocalityRegionalByRowCorrectZoneConfigBeforeBackfill.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/testutilsccl/BUILD.bazel
+++ b/pkg/ccl/testutilsccl/BUILD.bazel
@@ -10,6 +10,7 @@ go_library(
         "//pkg/base",
         "//pkg/jobs",
         "//pkg/roachpb",
+        "//pkg/settings/cluster",
         "//pkg/sql",
         "//pkg/sql/execinfra",
         "//pkg/sql/sqltestutils",

--- a/pkg/ccl/testutilsccl/alter_primary_key.go
+++ b/pkg/ccl/testutilsccl/alter_primary_key.go
@@ -17,6 +17,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/base"
 	"github.com/cockroachdb/cockroach/pkg/jobs"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
 	"github.com/cockroachdb/cockroach/pkg/sql"
 	"github.com/cockroachdb/cockroach/pkg/sql/execinfra"
 	"github.com/cockroachdb/cockroach/pkg/sql/sqltestutils"
@@ -62,14 +63,20 @@ func AlterPrimaryKeyCorrectZoneConfigTest(
 		t.Run(tc.Desc, func(t *testing.T) {
 			var db *gosql.DB
 			var params base.TestServerArgs
-			params.DefaultTestTenant = base.TestDoesNotWorkWithSharedProcessModeButWeDontKnowWhyYet(
-				base.TestTenantProbabilistic, 113853, /* issueNumber */
-			)
+			params.Settings = cluster.MakeClusterSettings()
 			params.Locality.Tiers = []roachpb.Tier{
 				{Key: "region", Value: "ajstorm-1"},
 			}
 
 			runCheck := false
+
+			// This setting must be overridden so that secondary tenants can configure
+			// regions.
+			sql.SecondaryTenantsMultiRegionAbstractionsEnabled.Override(
+				ctx,
+				&params.Settings.SV,
+				true,
+			)
 			params.Knobs = base.TestingKnobs{
 				SQLSchemaChanger: &sql.SchemaChangerTestingKnobs{
 					BackfillChunkSize: chunkSize,

